Add bower support

glepretre opened the pull request:
Closes #5.

@gentooboontoo @limrv @yoyey @robert19880720 Could you review this please? :blush:

glepretre committed:
Include scripts to source control

glepretre committed:
Use npm to manage dev dependencies

glepretre committed:
Add JSHint configuration file

glepretre committed:
Add JSCS configuration file

glepretre committed:
Use bower to manage dependencies

glepretre committed:
Rename service to distinguish it from module

glepretre committed:
Move pubsub core into its own module

glepretre committed:
Comply to npm test command convention

glepretre committed:
Update README

glepretre committed:
Increment major version number

glepretre committed:
Remove useless comment

glepretre committed:
Add comment about UMD pattern

gentooboontoo commented on README.md:
```diff
@@ -3,66 +3,112 @@ angular-pubsub
 * unsubscribe()
 
-##subscribing to a topic
+## subscribing to a topic
```
**S**ubscribing

gentooboontoo commented on README.md:
```diff
@@ -3,66 +3,112 @@ angular-pubsub
-##subscribing to a topic
+## subscribing to a topic
 
 The client (controller in most cases) which need to receive messages have to
```
need**s**

gentooboontoo commented on README.md:
```diff
@@ -3,66 +3,112 @@ angular-pubsub
+
 
 - 'topic' must be a string describing a topic/event (BUS name)
 - 'callback' must be a function, the callback executed when receiving a message
```
the callback **is** executed

gentooboontoo commented on README.md:
```diff
@@ -3,66 +3,112 @@ angular-pubsub
-##unsubscribing from a topic
+## unsubscribing from a topic
 
 When the client don't need to receive messages anymore, and to avoid memory leaks,
```
When the client **does not** need

gentooboontoo commented on README.md:
```diff
@@ -3,66 +3,112 @@ angular-pubsub
+## unsubscribing from a topic
 
 When the client don't need to receive messages anymore, and to avoid memory leaks,
 it is recommanded to unsubscribe
```
recomm**e**nded

gentooboontoo commented on Makefile:
```diff
@@ -0,0 +1,23 @@
+	rm --force --recursive ./$(OUTDIR)/*
+
+bower-install: bower.json
+	rm -rf $(BOWER_COMPONENTS_DIR)
```
--force --recursive

gentooboontoo commented on bower.json:
```diff
@@ -0,0 +1,45 @@
+  "license": "MIT",
+  "homepage": "https://github.com/glepretre/angular-pubsub",
+  "dependencies": {
+    "angular": "1.3.0"
```
Loosen dependency version.

gentooboontoo commented on src/angular-pubsub.js:
```diff
@@ -1,33 +1,30 @@
+    define(['angular', 'pubsub-core'], factory);
   }
-});
+  else if (typeof module === 'object' && module.exports) {
```
Useless for angular projects

gentooboontoo commented on src/angular-pubsub_test.js:
```diff
@@ -0,0 +1,176 @@
+        PubSub.publish(topic);
+
+        registeredCallbacks.forEach(function(callback) {
+          expect(callback).toHaveBeenCalled();
```
Useless?

gentooboontoo commented on src/angular-pubsub_test.js:
```diff
@@ -0,0 +1,176 @@
+        }
+
+        registeredCallbacks.forEach(function(callback) {
+          expect(callback).toHaveBeenCalled();
```
Useless ?

gentooboontoo commented:
Except comments, LGTM.

glepretre committed:
Capitalize README subtitles

glepretre committed:
Use explicit command options

glepretre committed:
Fix grammar

glepretre committed:
Use JSDoc syntax

glepretre committed:
Fix methods formatting

glepretre committed:
Loosen angular dependency version

limrv commented:
Same remarks as gentooboontoo for the utility of some specs? Otherwise LGTM.

glepretre committed:
Reduce angular service module defintion to AMD Web

As far as we know, angular and angular components do not run within
node environment.

glepretre committed:
Remove duplicate test expectation

Expecting a specific number of call count should be sufficient

glepretre committed:
Disable build minification
